packages/viz/src/CartoTileLayer.ts

```diff
@@ -1,88 +1,84 @@
 import { Credentials } from '@carto/toolkit-core';
 import { Maps } from '@carto/toolkit-maps';
-// import Style from './Style';
+import { GeoJsonLayer, MVTTileLayer } from '@deck.gl/layers';
 
-declare global {
-  interface Window {
-      deck: any;
-  }
-}
+import Source from './Source';
 
-class CartoTileLayer {
-  private _source: string;
+const defaultMapOptions: MapOptions = {
+  vector_extent: 2048,
+  vector_simplify_extent: 2048
+};
 
-  private _deck: any;
-  private _mapOptions: object;
+class CartoTileLayer {
+  private _mapsClientInstance: Maps;
   private _credentials: Credentials;
-  // private _style: object;
 
-  private _mapsClient: Maps;
+  private _mapSource: Source;
+  private _mapOptions: MapOptions;
+  private _mapInstantiation: Promise<any>; // TODO: Change to a proper definition
 
-  constructor(
-    source: string,
-    options: {
-      deckInstance?: any,
-      mapOptions?: object,
-      credentials?: Credentials
-    } = {}) {
+  constructor(source: string, options: LayerOptions = {}) {
+    const { mapOptions, credentials = new Credentials('jbotella', 'default_public') } = options;
 
-    this._source = source;
+    this._credentials = credentials;
+    this._mapsClientInstance = new Maps(this._credentials);
+    this._mapSource = new Source(source);
 
-    const { deckInstance, mapOptions, credentials } = options;
+    // Map Instantiation
+    this._mapOptions = Object.assign({}, defaultMapOptions, mapOptions);
+    this._mapInstantiation = this._mapsClientInstance.instantiateMapFrom(
```

Comment:
It's fine to have the map instantiation here 👍 
One relevant thing to consider is (if / when) we will need to 're-instantiate' a map. For example, will we have a 'setSource' to allow updating the Layer?